text/0000-project-descriptor-images.md

```diff
@@ -0,0 +1,159 @@
+# Meta
+[meta]: #meta
+- Name: Support specifying images in project descriptor
+- Start Date: 2021-10-26
+- Author(s): [@phil9909](https://github.com/phil9909), [@modulo11](https://github.com/modulo11), [@pbusko](https://github.com/pbusko), [@jkutner](https://github.com/jkutner)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+Add a new array of tables `[[images]]` to the `project.toml` file.
+
+# Definitions
+[definitions]: #definitions
+
+* **project descriptor**: The `project.toml` file that describes a project.
+* **image**: the output of a running a buildpack(s) against a project
+* **image name**: fully qualified name of an OCI image in the form `<registry>/<name>:<tag>`.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, there is no support for specifying an output image in the `project.toml` although flags like `--tag` are already present.
+
+# What it is
+[what-it-is]: #what-it-is
+
+Buildpack user will be able to specify the resulting images in the `project.toml`:
+
+# How it Works
+[how-it-works]: #how-it-works
+
+- Accepted project descriptor related RFCs
+  - https://github.com/buildpacks/rfcs/blob/main/text/0019-project-descriptor.md
+  - https://github.com/buildpacks/rfcs/blob/main/text/0054-project-descriptor-schema.md
+  - https://github.com/buildpacks/rfcs/blob/main/text/0080-builder-key-in-project-descriptor.md (could serve as a role-model)
+  - https://github.com/buildpacks/rfcs/blob/main/text/0084-project-descriptor-domains.md
+  - Closed PR which includes some similar changes: https://github.com/buildpacks/rfcs/blob/project-descriptor-simple/text/0000-project-descriptor.md
+
+Example:
+
+```toml
+[[images]]
+name = "spring-petclinic"
+registry = "gcr.io"
+tags = ["latest", "v1"]
+
+[[images]]
+name = "spring-petclinic"
+tags = ["latest", "v1"]
+
+[[images]]
+name = "spring-petclinic"
+registry = "private.registry.corp:8443"
```

Comment:
I wonder if we'd eventually want to add things like a `processes` table under this, or at least allow a `default-process` field.

Reply:
I'd like to see a user provided processes table, but definitely think that's out of scope for now.

If the images have different processes would that impact the way the image is built?